### DISCUSSION TITLE
Task.GetRunTimes start and end times

### DISCUSSION THREAD

**Discussion by ghost** [dahall/TaskScheduler#963] [Q&A] [answered]
For this method, is the assumption that the times passed are local times or UTC times? 

I have a test task set to start at midnight and runs every hour. If I run the following code:

```
TaskService ts = TaskService.Instance;
var x = ts.GetTask("TestTask");
DateTime s = new(2023, 5, 1, 0, 0, 0, DateTimeKind.Utc);
DateTime e = new(2023, 5, 2, 0, 0, 0, DateTimeKind.Utc);
DateTime[] z = x.GetRunTimes(s, e);
```

The times I get are not the expected times. Instead of starting with 5/1/2023 01:00, it starts with 4/30/2023 20:00 and gives me the expected 24 run times. This happens even if I check the "Synchronize across time zones" checkbox in Task Scheduler.

If I create the DateTime objects without the kind parameter, then update them with `s = s.ToUniversalTime()` and `e = e.ToUniversalTime()` then I get the expected results.

What am I doing wrong?

**Comment by dahall** [comment, marked as answer]
The `DateTime` values returned should all be in `DateTimeKind.Local`. An automatic conversion is done for the input values to local time, so it shouldn't matter their kind.
